### PR TITLE
Add a notebook to merge hifld licensed beds to hcris staffed ICU beds data

### PR DESCRIPTION
**Pull request by aaronxsu**
This PR adds a notebook to:
- download `HIFLD` licensed bed count data
- spatially join the above data with `HCRIS` data (which has facility level staffed ICU bed count, (with 150 meters buffer))
- deduplicate the joined result by checking address similarity and facility name similarity
- populate the non-matching records' licensed bed count by the estimated state average ratio of staffed ICU bed count to licensed bed count
- remove redundant fields and prefix fields from HIFLD data with `hifld_`
- export the merged and cleaned result to a `usa_hospital_beds_hcris2018_merge_hifld.geojson` file containing all `6,661` facilities originally from `HCRIS` data

Closes https://github.com/daveluo/covid19-healthsystemcapacity/issues/14
Closes https://github.com/daveluo/covid19-healthsystemcapacity/issues/25

### CODE REVIEW

File: nbs/hifld-licensed-bed-counts-for-all-US-health-facilities.ipynb [hunk 1]
```diff
@@ -0,0 +1,806 @@
...
+    "hifld_df.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hifld_gdf = gpd.GeoDataFrame(hifld_df, crs='EPSG:4326', geometry=gpd.points_from_xy(hifld_df.X, hifld_df.Y))"
```
Comment (lossyrob) [inline nbs/hifld-licensed-bed-counts-for-all-US-health-facilities.ipynb]: I needed to put 

```python
hifld_df = hifld_df.astype({'X': 'float32', 'Y': 'float32'})
```

before this line in order to make it work.

File: nbs/hifld-licensed-bed-counts-for-all-US-health-facilities.ipynb [hunk 2]
```diff
@@ -0,0 +1,806 @@
...
+    "len(joined_no_null_deduped)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "joined_no_null_deduped[\"icu_to_licensed\"] = joined_no_null_deduped[\"ICU Total Beds\"] / joined_no_null_deduped[\"hifld_BEDS\"]"
```
Comment (lossyrob) [inline nbs/hifld-licensed-bed-counts-for-all-US-health-facilities.ipynb]: Needed to set the dtype explicitly on `hifld_BEDS` to make this work:

```python
joined_no_null_deduped["icu_to_licensed"] = joined_no_null_deduped["ICU Total Beds"] / joined_no_null_deduped["hifld_BEDS"].astype('float32')
```

**Comment by lossyrob** [comment]
While running this I had to make a few changes, which I'll place as comments on the PR in case they are helpful. Also I merged in master which has some docker changes and not sure if that would effect things; I recommend doing that to ensure we're in the same latest environment.

I also hit a weird error on the last line saving to GeoJSON (I got `TypeError: data type not understood`).

My immediate thought to get around this, as I'm not a geopandas person, is to use 

```python
result.to_dict(orient='record')
```
And build up the GeoJSON manually via `dict`s.

Maybe @simonkassel would know better?

**Comment by simonkassel** [comment]
@lossyrob @aaronxsu  Ok, I think I figured out the issue. The problem is that GeoPandas can only handle one geometry column. It understands `geometry` to be the "geometry" attribute of the GDF but doesn't know how to serialize the shapely polygons in `geom_buffered`. You could either drop `geom_buffered` and then write off the geojson or convert it to a dict first then write it off.

```
result['geom_buffered'] = result['geom_buffered'].apply(lambda x: x.__geo_interface__)
result.to_file('usa_hospital_beds_hcris2018_merge_hifld.geojson', encoding='utf-8', driver='GeoJSON')
```
I find that the geopandas `to_dict()` method usually doesn't give me the data in a format that I want and doesn't convert the shapely geoms to a dict. When I want to go from geopandas to GeoJSON i use the `.to_json()` method which returns a geojson string that you can either write off as a file or parse into a python dict. 

```
import json
json.loads(result.to_json())
```

But either way you will need to manually convert the second column of shapely geoms to a geojson dict first or you will get an error.

**Comment by aaronxsu** [comment]
@simonkassel @lossyrob YES! Thanks so much for helping me figure it out! I should have looked at here earlier... I found out the problem in the past hour too. Please find the resulting data from the latest commit.

**Comment by lossyrob** [comment]
This looks good! I like the use of name similarity to deduplicate the matches.

I have a PR that makes some tweaks here https://github.com/aaronxsu/covid19-healthsystemcapacity/pull/1

Otherwise I think this is good to go!

**Comment by lossyrob** [comment]
Thank you!

**Pull request merged**